src/enlyze/client.py

```diff
@@ -139,56 +156,107 @@ def get_timeseries(
 
         :raises: |generic-error|
 
-        :returns: Timeseries data or ``None`` if API returned no data for the request
+        :returns: Timeseries data or ``None`` if the API returned
+            no data for the request
 
         """
 
-        if start.utcoffset() is None or end.utcoffset() is None:
-            logging.warning(
-                "Passing naive datetime is discouraged, assuming local timezone."
-            )
+        start, end, appliance_uuid = validate_timeseries_arguments(
+            start, end, variables
+        )
+
+        pages = self._client.get_paginated(
+            "timeseries",
+            api_models.TimeseriesData,
+            params={
+                "appliance": appliance_uuid,
+                "start_datetime": start.isoformat(),
+                "end_datetime": end.isoformat(),
+                "variables": ",".join(str(v.uuid) for v in variables),
+            },
+        )
+
+        timeseries_data = _get_timeseries_data_from_pages(pages)
+        if timeseries_data is None:
+            return None
+
+        return timeseries_data.to_user_model(
+            start=start,
+            end=end,
+            variables=variables,
+        )
+
+    def get_timeseries_with_resampling(
+        self,
+        start: datetime,
+        end: datetime,
+        variables: list[user_models.VariableWithResamplingMethod],
+        resampling_interval: int,
```

Comment:
I don't understand why we'd use `datetime.timedelta`. 

>  it's not clear that this is expected to be in seconds.

Fixed in the docstring.

Reply:
The problem with using `timedelta` is that it's not very ergonomic for us when it comes to negative quantities

```
>>> td = timedelta(seconds=-5)
>>> td
datetime.timedelta(days=-1, seconds=86395)
>>> td.seconds
86395
```

What should our approach be there?

Reply:
There's an API for exactly that :upside_down_face: 

```
>>> datetime.timedelta(days=-1, seconds=86395).total_seconds()
-5.0
```

All we'll have to do is to convert it to `int` afterwards (silently dropping / rounding any milliseconds)

Reply:
Exactly, I am aware of `total_seconds`, the thing is, all this gymnastics is useless to us because we don't do anything interesting with `timedelta`, we just need an int, validate it against another int, and send it in a request. All this added complexity in logic is useless, and introduces extra caveats, e.g. if the user sees they're allowed to pass a `timedelta`, then are microseconds and milliseconds valid or not? We know they aren't but we need to communicate that to the user. Judging from all of this, this is not a useful change and under the presented rationale, I am not convinced of any value offered.